src/entities/BlockNode/BlockNode.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { TextNode } from '../TextNode';
+import { ValueNode } from '../ValueNode';
+
+import type { EditorDocument } from '../EditorDocument';
+import type { BlockTuneConstructorParameters } from '../BlockTune/types';
+import type { TextNodeConstructorParameters } from '../TextNode';
+import type { ValueNodeConstructorParameters } from '../ValueNode';
+import type { InlineNodeSerialized } from '../interfaces';
+
+describe('BlockNode', () => {
+  describe('.serialized', () => {
+    it('should return a name of a tool that created a BlockNode', () => {
+      const blockNodeName = createBlockNodeName('paragraph');
+
+      const blockNode = new BlockNode({
+        name: blockNodeName,
+        data: {},
+        parent: {} as EditorDocument,
+      });
+
+      const serialized = blockNode.serialized;
+
+      expect(serialized.name).toEqual(blockNodeName);
+    });
+
+    it('should call .serialized getter of all tunes associated with the BlockNode', () => {
+      const blockTunesNames = [
+        createBlockTuneName('align'),
+        createBlockTuneName('font-size'),
+        createBlockTuneName('font-weight'),
+      ];
+
+      const spy = jest
+        .spyOn(BlockTune.prototype, 'serialized', 'get')
+        .mockImplementation(() => ({} as BlockTuneSerialized));
+
+      const blockNode = new BlockNode({
+        name: createBlockNodeName('paragraph'),
+        data: {},
+        parent: {} as EditorDocument,
+        tunes: blockTunesNames.reduce((acc, name) => ({
+          ...acc,
+          [name]: new BlockTune({} as BlockTuneConstructorParameters),
+        }), {}),
+      });
+
+      blockNode.serialized;
+
+      expect(spy).toHaveBeenCalledTimes(blockTunesNames.length);
+    });
+
+    it('should call .serialized getter of all nodes associated with the BlockNode', () => {
+      const textNodeSerializedSpy = jest
+        .spyOn(TextNode.prototype, 'serialized', 'get')
+        .mockImplementation(() => ({} as InlineNodeSerialized));
+
+      const valueNodeSerializedSpy = jest
+        .spyOn(ValueNode.prototype, 'serialized', 'get')
+        .mockImplementation(() => ({}));
+
+      const countOfTextNodes = 3;
+      const countOfValueNodes = 1;
+
+      const textNodes = [ ...Array(countOfTextNodes).keys() ]
+        .map((index) => new TextNode({ value: `${index}` } as TextNodeConstructorParameters));
+
+      const valueNodes = [ ...Array(countOfValueNodes).keys() ]
+        .reduce((acc, index) => ({
+          ...acc,
+          [createDataKey(`data-key-${index}c${index}d`)]: new ValueNode({} as ValueNodeConstructorParameters),
+        }), {});
+
+      const blockNode = new BlockNode({
+        name: createBlockNodeName('paragraph'),
+        data: {
+          [createDataKey('data-key-1a2b')]: textNodes,
+          ...valueNodes,
+        },
+        parent: {} as EditorDocument,
+      });
+
+      blockNode.serialized;
+
+      expect(textNodeSerializedSpy).toHaveBeenCalledTimes(countOfTextNodes);
```

Comment:
Would be better to split this case into two: one for value nodes, one for text nodes